Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -1187,6 +1186,7 @@ private void HandleOrderEvents(List<OrderEvent> orderEvents)
 
                     // update the ticket after we've processed the fill, but before the event, this way everything is ready for user code
                     tickets[i].AddOrderEvent(orderEvent);
+                    orderEvent.Ticket = tickets[i];
```

Comment:
I think we might be able to get rid of the `tickets` collection entirely here, maybe we just set the ticker on the order even on the first loop at the top?

Reply:
Also, when we worked on HandleOrderEvents, we needed to make sure to find every order and ticket before starting to update the orders, that being the reason of having separate loops and the collections.